### PR TITLE
New timer architecture using BukkitScheduler

### LINKED ISSUE #17: Reimplement timer using BukkitScheduler
Currently, the timer runs using a separate timer thread that sleeps repeatedly. However, this basically amounts to busy-waiting, however efficient it is, and it would be better to delegate the timing logic to the scheduler instead. Let's rewrite the timer code to instead use the BukkitScheduler and take advantage of its timing functionalities.

### PR DESCRIPTION
**Pull request by Aechylus**
Implementation for issue #17

### CODE REVIEW

File: src/com/clancraft/turnmanager/turn/TurnTimer.java [hunk 1]
```diff
@@ -1,110 +1,106 @@
...
-            try {
-                Thread.sleep(TIMER_RESOLUTION_MINS * MIN_TO_MS);
-                minutesRemaining -= TIMER_RESOLUTION_MINS;
+        // TODO check if the modulo math is correct.
+        normalBroadcastId = Bukkit.getScheduler().scheduleSyncRepeatingTask(plugin,
+                () -> {
+                    long minsRemaining = TimeUnit.MILLISECONDS.toMinutes(deadlineTimeMillis - System.currentTimeMillis());
+                    Bukkit.broadcastMessage(String.format(TMConstants.TIMER_COUNTDOWN, minsRemaining));
+                },
+                timerDurationMillis % TimeUnit.MINUTES.toMillis(NORMAL_INTERVAL_MINS),
```
Comment (Phobez) [inline src/com/clancraft/turnmanager/turn/TurnTimer.java]: This parameter is `delay - Delay in server ticks before executing first repeat` right? Why not just `TimeUnit.MINUTES.toMillis(NORMAL_INTERVAL_MINS)`?

**Review by Phobez — CHANGES_REQUESTED** [review]

File: src/com/clancraft/turnmanager/turn/TurnTimer.java (hunk 1 above)
Comment (Aechylus) [reply src/com/clancraft/turnmanager/turn/TurnTimer.java]: That's so the timer runs on multiples of NORMAL_INTERVAL_MINS. Say you have 17 mins left. I want the first broadcast to be 2 mins in the future, saying "Timer has 15 mins left".

Comment (Phobez) [reply src/com/clancraft/turnmanager/turn/TurnTimer.java]: Ah, I see. Got it then

Comment (Aechylus) [reply src/com/clancraft/turnmanager/turn/TurnTimer.java]: But anyways do you think it's a good idea to immediately broadcast timer on resumption with whatever values it has? Like a one-time "Timer has 17 mins left" and subsequently just the multiples of 5 mins?

**Review by Phobez — APPROVED** [review]

File: src/com/clancraft/turnmanager/turn/TurnTimer.java (hunk 1 above)
Comment (Phobez) [reply src/com/clancraft/turnmanager/turn/TurnTimer.java]: Yeah I think that sounds fine. Are you going to add it to this PR as well?

Comment (Aechylus) [reply src/com/clancraft/turnmanager/turn/TurnTimer.java]: I think so, and I'll add comments explaining what I did above too

Comment (Phobez) [reply src/com/clancraft/turnmanager/turn/TurnTimer.java]: Alright then I'll hold off on merging

**Comment by Aechylus** [comment]
Change is more extensive than I thought alas, but should be good now.

**Review by Phobez — APPROVED** [review]

**Pull request merged**